freedomofpress/securedrop-client
# Disable download when conversation was deleted

# Description

:crystal_ball: **Reviewers** (@creviera): This PR builds on #1447, please wait until that one is merged to rebase then review. Only two commits will remain :slightly_smiling_face: (In the meantime you can look at the last two commits to get an idea of the changes involved.)

Fixes #1442 

# Test Plan

Ensure that the steps to reproduce defined in #1442 are handled:

- Select a source with files that have not yet been downloaded
- Delete the source conversation
- Select the source menu
- [ ] Confirm that the "Download All Files" action is disabled

# Comments

This PR gives `DeleteConversationAction` access to the application state, which means we can now expand `state.State` to contain information about messages and use it to enable/disable the action dynamically. It's out of scope, but I thought it was worth mentioning.

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
